### PR TITLE
Handle add account messages.

### PR DESCRIPTION
**Pull request by smarlowucf**
- Process add account request messages in credentials service.
- Accounts are added to json file and the credentials are encrypted and stored.
- Update unit tests for account handling changes.

### CODE REVIEW

File: mash/services/credentials/base_account.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        Add the account to the group for the requesting user.
+
+        If the group does not exist create it with the new account.
+        """
+        groups = accounts[self.provider]['groups'].get(self.requesting_user)
+
+        if groups:
+            group = groups.get(self.group_name)
+
+            if group is None:
```
Comment (schaefi) [inline mash/services/credentials/base_account.py]: Could be changed to

```python
if not group:
```

Comment (schaefi) [reply mash/services/credentials/base_account.py]: I think empty strings are also considered a no group which are not matched when looking for None type

File: test/unit/services/credentials/base_account_test.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    def setup(self):
+        self.account = BaseAccount(
+            'acnt123', 'aws', 'ec2', 'user3', group_name='group123'
+        )
+
+    def test_add_account(self):
+        self.account.add_account({'test': 'data'})
+
+    def test_add_multiple_accounts(self):
+        with open('../data/accounts.json') as f:
```
Comment (schaefi) [inline test/unit/services/credentials/base_account_test.py]: nitpick but ```f``` is a bad name :smile:

File: mash/services/credentials/base_account.py (hunk 1 above)
Comment (smarlowucf) [reply mash/services/credentials/base_account.py]: Yeah, this would only return a dict or None but that works.

**Comment by smarlowucf** [comment]
@schaefi Made the suggested changes, going ahead with merge.

**Pull request merged**